### ISSUE TITLE
interaction of system storage config with instance can be very confusing

### ISSUE THREAD

**Issue by mgasner** [dagster-io/dagster#2259]
system_storage.py l. 59 et seq

if you have config like

```
storage:
  filesystem:
    config:
      base_dir: 'whatever'
```

then you end up with the potentially unexpected `root_for_run_id=lambda _: override_dir`

**Comment by mgasner** [comment]
i hit this in tests today, with a setup like:

```
    with seven.TemporaryDirectory() as temp_dir:
        instance = DagsterInstance.local_temp(temp_dir)

        ref = instance.get_ref()

        def subprocess_target(ref):
            execute_pipeline(
                foo_pipeline,
                {
                    ...,
                    'storage': {'filesystem': {'config': {'base_dir': temp_dir}}}
                },
                instance=DagsterInstance.from_ref(ref)
            )
```

**Comment by asharov** [comment]
I'm hitting this too. I have multiple pipeline repositories and I've configured `intermediate_storage` with `base_dir` to have all the intermediate storages for them under the same root directory. But I noticed that the run id is not part of a run's intermediate storage path, making it impossible to rerun parts of old runs for instance, and that looks like this issue.

I'm still on Dagster 0.9. Will this issue be fixed in some way with the IO managers of 0.10?

**Comment by sryza** [comment]
Hi @asharov - the IO managers in 0.10.0 should not have this problem.

You should be able to achieve what you want with:
```
@pipeline(mode_defs=[ModeDefinition(resource_defs={"io_manager": fs_io_manager.configured("base_dir": "your_base_dir")})])
    ...
```
